lib/elixir/lib/macro.ex

```diff
@@ -370,6 +370,25 @@ defmodule Macro do
       iex> Macro.escape({:unquote, [], [1]}, unquote: true)
       1
 
+  The `Macro.escape/2` function is sometimes confused with `Kernel.quote/2`,
+  because the above examples behave the same with both. The key difference is
+  best illustrated when the value to escape is stored in a variable.
+
+  ## Comparison to `Kernel.quote/2`
+
+      iex> quote do: {:a, :b, :c}
+      {:{}, [], [:a, :b, :c]}
+
+      iex> value = {:a, :b, :c}; Macro.escape(value)
```

Comment:
Please do this instead of putting two expressions on the same line and using `;`:

```elixir
iex> value = {:a, :b, :c}
iex> Macro.escape(value)
```

here and in other places.

Reply:
https://github.com/elixir-lang/elixir/pull/7407/files#r171648405